man/systemd.exec.xml

```diff
@@ -1129,6 +1129,18 @@ BindReadOnlyPaths=/var/lib/systemd</programlisting>
         security.</para></listitem>
       </varlistentry>
 
+      <varlistentry>
+        <term><varname>ProtectHostname=</varname></term>
+
+        <listitem><para>Takes a string argument. When set, sets up a new UTS namespace for the executed
+        processes and changes hostname to the argument with usual <literal>%</literal>-specifiers
+        resolved. In addition, changing hostname is prevented. Defaults to off.</para>
```

Comment:
hmm, i would have assume this would be simple boolean, and the hostname set be the same as the one from the host. Is there any usecase you are aware of for making the hostname configurable? I'd see this as a sandboxing option, that takes privs away, not as a way to give processes their own view of the world after all...

Reply:
Then the UTS namespaces would be pretty much useless. I'd then just extend coverage of `ProtectKernelTunables` to `/proc/sys/kernel/{host,domain}name`. The system calls can be prevented with simple `SystemCallFilter`, maybe a new syscall group could be added or one of the existing could be extended.

Reply:
ProtectKernelTunables already protects those files (as it protects the whole of /proc/sys/kernel/).

I'd see the ProtectHostname= as a more finegrained operation: it only protects one very specific (though prominent) kernel setting.

The thing with ProtectKernelTunables= is that it works with ro-bind mounts, and they can be easily undone unless combined with restrictions on mount(). Which is the reason why the UTS namespacing is interesting here: it means that undoing the r/o mounting means you gained the ability to change the hostname your service sees, but the host's hostname is still off limits. 

Hence, the combination of the three restrictions is pretty interesting I'd say, as each restriction individually leaves a gaping hole, but all three in combination are substantially more secure. If that makes sense?